cache/util/fsutil.go

```diff
@@ -131,16 +131,23 @@ func StatFile(ctx context.Context, mount snapshot.Mountable, path string) (*fsty
 			// The filename here is internal to the mount, so we can restore
 			// the request base path for error reporting.
 			// See os.DirFS.Open for details.
-			err1 := err
-			if err := errors.Cause(err); err != nil {
-				err1 = err
-			}
-			if pe, ok := err1.(*os.PathError); ok {
-				pe.Path = path
-			}
+			setErrorPath(err, path)
 			return errors.WithStack(err)
 		}
 		return nil
 	})
 	return st, err
 }
+
+// setErrorPath will override the path in an os.PathError in the error chain.
+// This works with the fsutil library, but it isn't necessarily the correct
+// way to do this because the error message of wrapped errors doesn't necessarily
+// update or change when a wrapped error is changed.
+//
+// Still, this method of updating the path works with the way this specific
+// library returns errors.
+func setErrorPath(err error, path string) {
+	if pe := (*os.PathError)(nil); errors.As(err, &pe) {
```

Comment:
Is this `nil` casting something that is a common pattern? I understand it works but looks quite weird. Usually this is written as

```
var pe *os.PathError
if errors.As(err, &pe) {}
```

Reply:
I don't know if it's a common pattern, but I've generally been annoyed that the typical way causes the variable to leak into another scope. I can change it to the more typical way. I just like to try and keep variable scopes as constrained as I can in most circumstances.